File: backend/models/Payment.js
// backend/models/Payment.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

const Payment = sequelize.define('Payment', {
  payment_id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  reservation_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  payer_user_id: {
    type: DataTypes.STRING(50),
    allowNull: false
  },
  payment_type: {
    type: DataTypes.ENUM('DEPOSIT', 'BALANCE', 'REFUND'),
    allowNull: false
  },
  status: {
    type: DataTypes.ENUM('AUTHORIZED', 'CAPTURED', 'CANCELED', 'REFUNDED', 'FAILED'),
    defaultValue: 'AUTHORIZED'
  },
  method: {
    type: DataTypes.STRING(50),
    allowNull: true
  },
  provider: {
    type: DataTypes.STRING(50),
    allowNull: true
  },
  provider_txn_id: {
    type: DataTypes.STRING(200),
    allowNull: true
  },
  amount: {
    type: DataTypes.DECIMAL(12, 2),
    allowNull: false,
    validate: {
      min: 0
    }
  },
  currency: {
    type: DataTypes.STRING(3),
    defaultValue: 'KRW'
  },
  related_payment_id: {
    type: DataTypes.INTEGER,
    allowNull: true
  },
  paid_at: {
    type: DataTypes.DATE,
    allowNull: true
  }
}, {
  tableName: 'payments',
  timestamps: true,
  underscored: true,
  createdAt: 'created_at',
  updatedAt: 'updated_at',
  indexes: [
    {
      fields: ['reservation_id', 'payment_type', 'status']
    }
  ]
});

module.exports = Payment;

File: spring-backend/src/main/java/com/noshow/app/domain/entity/Payment.java
package com.noshow.app.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
  public enum PaymentType { DEPOSIT, BALANCE, REFUND }
  public enum Status { AUTHORIZED, CAPTURED, CANCELED, REFUNDED, FAILED }

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "payment_id")
  private Long paymentId;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "reservation_id", nullable = false)
  @ToString.Exclude
  private Reservation reservation;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "payer_user_id", nullable = false)
  private User payer;

  @Enumerated(EnumType.STRING)
  @Column(name = "payment_type", nullable = false)
  private PaymentType paymentType;

  @Column(name = "method", nullable = false, length = 30)
  private String method;

  @Column(name = "provider", length = 30)
  private String provider;

  @Column(name = "provider_txn_id", length = 100)
  private String providerTxnId;

  @Column(name = "amount", nullable = false)
  private BigDecimal amount;

  @Column(name = "currency", nullable = false, length = 3)
  private String currency;

  @Enumerated(EnumType.STRING)
  @Column(name = "status", nullable = false)
  private Status status;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "related_payment_id")
  private Payment relatedPayment;

  @Column(name = "paid_at")
  private LocalDateTime paidAt;

  @CreationTimestamp
  @Column(name = "created_at", insertable = false, updatable = false)
  private LocalDateTime createdAt;

  @UpdateTimestamp
  @Column(name = "updated_at", insertable = false)
  private LocalDateTime updatedAt;
}
